src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -599,10 +601,12 @@ class NetworkOPsImp final : public NetworkOPs
 
     Json::Value
     transJson(
-        const STTx& transaction,
+        std::shared_ptr<STTx const> const& transaction,
         TER result,
         bool validated,
-        std::shared_ptr<ReadView const> const& ledger);
+        std::shared_ptr<ReadView const> const& ledger,
+        std::optional<std::reference_wrapper<TxMeta const>> meta,
```

Comment:
Since the `meta` parameter is only being read, its not being written into -- I attempted to remove the `std::reference_wrapper` [tip of this branch](https://github.com/ckeshava/rippled/tree/noRefWrap)
The unit tests are passing, but let me know the merits of using ref wrapper

Reply:
I use `reference_wrapper` because it's an equivalent of passing by reference, i.e. saves you copying an object (`TxMeta` contains a `std::vector<STObject>` encapsulated inside `STArray` - this could be potentially heavy to copy). Technically `reference_wrapper` is a safe wrapper for a non-owning pointer.

This parameter is optional, and you _cannot_ put a reference type inside `std::optional`, so the most natural parameter type `std::optional<TxMeta const&>` won't work. Your choices are either:
- express in code what your intent is (which is what `std::optional<std::reference_wrapper<...>>` does), or
- pass a raw pointer (which can be null)

I find passing raw pointers an anti-pattern because they have potentially so many uses (many of them bad) that one needs to wonder "why here ?" when reading code. Which of course slows people down and may lead potential troubleshooters in the future on a wild goose chase.

Reply:
ok this helps. thanks for the explanation